tests/integration/test_envelope.py

```diff
@@ -579,3 +579,51 @@ def test_sample_rates_metrics(mini_sentry, relay_with_processing, events_consume
 
     event, _ = events_consumer.get_event()
     assert event["_metrics"]["sample_rates"] == sample_rates
+
+
+def test_buffer_envelopes_without_global_config(
+    mini_sentry, relay_with_processing, events_consumer
+):
+    """
+    Checks that we buffer envelopes until we have a global config in processing mode.
+    """
+
+    include_global = False
+    original_endpoint = mini_sentry.app.view_functions["get_project_config"]
+
+    @mini_sentry.app.endpoint("get_project_config")
+    def get_project_config():
+        nonlocal include_global
+
+        res = original_endpoint().get_json()
+        if not include_global:
+            res.pop("global")
+        return res
+
+    mini_sentry.add_basic_project_config(42)
+    events_consumer = events_consumer()
+    options = {"cache": {"global_config_fetch_interval": 1}}
+    relay = relay_with_processing(options=options)
+
+    envelope_qty = 5
+    for _ in range(envelope_qty):
+        envelope = Envelope()
+        envelope.add_event({"message": "hello, world!"})
+        relay.send_envelope(42, envelope)
+
+    events_consumer.assert_empty()
```

Comment:
IIUC, this runs without any time delay. I have the same concern as in https://github.com/getsentry/relay/pull/2697#discussion_r1394280536.

Reply:
iirc it has an internal 1 second delay

Reply:
Actually I am wrong, it does **not** have an internal timeout of 1 second. 🚨

Reply:
yeah good catch! I added 2 sec sleep now

Reply:
that makes sense, thx! changed it